alert-database/src/main/java/com/synopsys/integration/alert/database/job/DistributionJobRepository.java

```diff
@@ -71,4 +71,58 @@
     )
     List<DistributionJobEntity> findMatchingEnabledJob(@Param("providerConfigId") Long providerConfigId, @Param("notificationType") String notificationType);
 
+    @Query(value = "SELECT DISTINCT jobEntity FROM DistributionJobEntity jobEntity "
+                       + "    LEFT JOIN jobEntity.blackDuckJobDetails blackDuckDetails ON jobEntity.jobId = blackDuckDetails.jobId "
+                       + "    LEFT JOIN blackDuckDetails.blackDuckJobNotificationTypes notificationTypes ON jobEntity.jobId = notificationTypes.jobId "
+                       + "    LEFT JOIN blackDuckDetails.blackDuckJobPolicyFilters policyFilters ON jobEntity.jobId = policyFilters.jobId "
+                       + "    LEFT JOIN blackDuckDetails.blackDuckJobVulnerabilitySeverityFilters vulnerabilitySeverityFilters ON jobEntity.jobId = vulnerabilitySeverityFilters.jobId "
+                       + "    LEFT JOIN blackDuckDetails.blackDuckJobProjects projects ON jobEntity.jobId = projects.jobId "
+                       + "    WHERE jobEntity.enabled = true"
+                       + "    AND notificationTypes.notificationType = :notificationType"
+                       + "    AND jobEntity.distributionFrequency = :frequency"
+                       + "    AND (blackDuckDetails.filterByProject = false OR blackDuckDetails.projectNamePattern IS NOT NULL OR projects.projectName = :projectName)"
+    )
+    List<DistributionJobEntity> findMatchingEnabledJobs(
```

Comment:
There's a pretty high chance we want to return a page of jobs here.

Also, you don't need to return the entire entity if you don't want to. You can change 
```sql
SELECT DISTINCT jobEntity FROM DistributionJobEntity jobEntity . . .
``` 
to 
```sql
SELECT DISTINCT new FilteredDistributionJobModel(jobEntity.processingType, jobEntity.jobId, jobEntity.channelDescriptorName) 
FROM DistributionJobEntity jobEntity . . .
```

Reply:
I'm not even sure I'd need DISTINCT here tbh. But will make this change